crates/pixi/tests/integration_rust/build_tests.rs

```diff
@@ -544,3 +545,106 @@ backend.version = "0.1.0"
         ".pixi/.gitignore file was not created after build"
     );
 }
+
+/// Test that demonstrates using PassthroughBackend with PixiControl
+/// to test build operations without requiring actual backend processes.
+#[tokio::test]
+async fn test_different_variants_have_different_caches() {
+    setup_tracing();
+
+    // Create a package database with common dependencies
+    let package_database = create_test_package_database();
+
+    // Convert to channel
+    let channel = package_database.into_channel().await.unwrap();
+
+    // Create a PixiControl instance with PassthroughBackend
+    // Create an observable backend and get the observer
+    let (instantiator, mut observer) =
+        ObservableBackend::instantiator(PassthroughBackend::instantiator());
+
+    let backend_override = BackendOverride::from_memory(instantiator);
+
+    let pixi = PixiControl::new()
+        .unwrap()
+        .with_backend_override(backend_override);
+
+    // Create a simple source directory
+    let source_dir = pixi.workspace_path().join("my-package");
+    fs::create_dir_all(&source_dir).unwrap();
+
+    // Create a pixi.toml that the PassthroughBackend will read
+    let pixi_toml_content = r#"
+[package]
+name = "my-package"
+version = "1.0.0"
+
+[package.build]
+backend = { name = "in-memory", version = "0.1.0" }
+
+[package.host-dependencies]
+sdl2 = "*"
+
+[package.build.config]
+on-the-fly = true
```

Comment:
Shall we just default to this if the package doesnt define an existing package? Then we can remove this.